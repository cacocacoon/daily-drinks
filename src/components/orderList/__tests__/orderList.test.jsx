import React from 'react'
import ReactDOM from 'react-dom'
import { act } from "react-dom/test-utils"
import { render } from '@testing-library/react'
import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { MemoryRouter, Route, Switch } from 'react-router-dom'

import OrderList from '../orderList'

afterEach(() => {
    // cleanup on exiting
    ReactDOM.unmountComponentAtNode(document.body);
});
const mockStore = configureMockStore()

function TestOrderList({ store }) {
    return (
        <Provider store={store}>
            <MemoryRouter>
                <Switch>
                    <Route path="/" component={OrderList} exact/>
                </Switch>
            </MemoryRouter>
        </Provider>
    )
}

describe('OrderList', () => {
    it('default view', () => {
        const store = mockStore({ orders: { list: [{
            name: 'iPhone X',
            price: '1099.99',
            checked: false,
            notes: `iPhone X classic case * 1`
        }, {
            name: 'Electric Bike E7',
            price: '2100',
            checked: true,
            notes: `if you never try, you never know`
        }] } })
        const orderDetail = TestRenderer.create(
            <TestOrderList store={store} />
        )
        expect(orderDetail.toJSON()).toMatchSnapshot()
    })

    it('empty view', () => {
        const store = mockStore({ orders: { list: [] } })
        const orderDetail = TestRenderer.create(
            <TestOrderList store={store} />
        )
        expect(orderDetail.toJSON()).toMatchSnapshot()
    })

    it('submit', () => {
        const store = mockStore({ orders: { list: [{
            name: 'iPhone X',
            price: '1099.99',
            checked: false,
            notes: `iPhone X classic case * 1`
        }] } })
        store.dispatch = jest.fn(() => {})
        let renderResults
        act(() => {
            renderResults = render(<TestOrderList store={store} />)
        })
        const checked = renderResults.container.querySelector('.checked')
        checked.dispatchEvent(new MouseEvent('click', { bubbles: true }))

        expect(store.dispatch.mock.calls[0][0]).toStrictEqual({ payload: true, type: "SHOW_LIST"})
        expect(store.dispatch).toHaveBeenCalledTimes(1)
    })
})