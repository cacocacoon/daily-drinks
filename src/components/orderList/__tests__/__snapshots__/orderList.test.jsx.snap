// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList default view 1`] = `
<div
  className="order-list"
>
  <div
    className="order-list-head"
  >
    <div
      className="checked"
    />
    <div
      className="name"
    >
      Name
    </div>
    <div
      className="price"
    >
      Price
    </div>
    <div
      className="notes"
    >
      Notes
    </div>
  </div>
  <a
    className="order"
    href="/order/0"
    onClick={[Function]}
  >
    <div
      className="checked"
      data-index={0}
      onClick={[Function]}
    >
      ○
    </div>
    <div
      className="name"
    >
      iPhone X
    </div>
    <div
      className="price"
    >
      1099.99
    </div>
    <div
      className="notes"
    >
      iPhone X classic case * 1
    </div>
  </a>
  <a
    className="order"
    href="/order/1"
    onClick={[Function]}
  >
    <div
      className="checked"
      data-index={1}
      onClick={[Function]}
    >
      ●
    </div>
    <div
      className="name"
    >
      Electric Bike E7
    </div>
    <div
      className="price"
    >
      2100
    </div>
    <div
      className="notes"
    >
      if you never try, you never know
    </div>
  </a>
</div>
`;

exports[`OrderList empty view 1`] = `
<div
  className="empty-order-list"
>
  Empty
</div>
`;
